c10/core/WrapDimMinimal.cpp

```diff
@@ -42,5 +33,15 @@ T maybe_wrap_dim_slow(T dim, T dim_post_expr, bool wrap_scalar) {
       false, "should never reach here as dim should be out-of-bounds");
 }
 
+// Explicitly instantiate the template at the two types it will be used
```

Comment:
I could guess why C10_API is important for linker errors across lib boundaries,

is moving it below also important for some reason or just conventional?

Reply:
nah, it's just cuz I feel it's weird to instantiate a template before you actualy defined it